packages/astro/src/assets/services/service.ts

```diff
@@ -216,6 +221,89 @@ export const baseService: Omit<LocalImageService, 'transform'> = {
 			decoding: attributes.decoding ?? 'async',
 		};
 	},
+	getSrcSet(options) {
+		const srcSet: SrcSetValue[] = [];
+		const { targetWidth, targetHeight } = getTargetDimensions(options);
+		const { widths, densities } = options;
+		const targetFormat = options.format ?? DEFAULT_OUTPUT_FORMAT;
+
+		const aspectRatio = targetWidth / targetHeight;
+		const imageWidth = isESMImportedImage(options.src) ? options.src.width : options.width;
+		const maxWidth = imageWidth ?? Infinity;
+
+		// REFACTOR: Could we merge these two blocks?
+		if (densities) {
```

Comment:
Firstly, thank you @Princesseuh for your work on this! Such a lovely feature. I've been using the latest previews here for a gallery of mine, just calling `getImage` to use the srcset generation, and it works like a charm. Just one issue that I've come across, but maybe I'm just using it incorrectly!

I think the conditional here for when either `densities` or `widths` is defined means that something like the following is not possible:

```astro
<Picture
  alt="An image in two formats with one width"
  src={image}
  width={700}
  formats={['avif', 'webp']}
/>
```

Which I would think could generate something like the following:

```html
<picture>
  <source srcset="/_astro/my_image_700w.hash.avif" type="image/avif" />
  <source srcset="/_astro/my_image_700w.hash.webp" type="image/webp" />
  <img
    src="/_astro/my_image_700w.hash.jpg"
    width="700"
    height="350"
    decoding="async"
    loading="lazy"
    alt="An image in two formats with one width"
  />
</picture>
```

Where it wouldn't generate a srcset like `[url] [width]w, [url] [width]w`, etc. but instead as the single resized image.

Instead, it results in this error:

```
error   Cannot read properties of undefined (reading 'attributes')
Code:
  51 |         srcset={`${image.src}, ` + image.srcSet.attribute}
> 52 |         type={image.srcSet.values[0].attributes?.type}
     |                                      ^
```

Basically, I think that it's getting to this point and not generating any srcSet values. If instead I defined it as `widths={[700]}` it would produce the 700w image and the original size in the srcset, when I'd just like to resize the image to 700px and have multiple file types.

Is this a bug or is there another way to achieve this with this impl? Maybe just out of scope?

Reply:
Fixed in the latest commit, sorry for the inconvenience!

The problem wasn't the block of code you shared, but the fact that we used the `srcset` generated to find the attributes to use on the different formats, so when you had no srcset (like you shared, only one width), we wouldn't be able to find the proper attributes.